python/runtime/feature_derivation/feature_column.py

```diff
@@ -0,0 +1,512 @@
+            Currently 'mean', 'sqrtn' and 'sum' are supported.
+        initializer (str): the initializer of the embedding table.
+        name (str): only used when category_column=None. In this case, the
+            category_column would be filled automaticaly in the feature derivation
+            stage.
+    """
+    def __init__(self,
+                 category_column=None,
+                 dimension=0,
+                 combiner="",
+                 initializer="",
+                 name=""):
+        if category_column is not None:
+            assert isinstance(category_column, CategoryColumn)
+
+        self.category_column = category_column
+        self.dimension = dimension
+        self.combiner = combiner
+        self.initializer = initializer
+        self.name = name
+
+    def get_field_desc(self):
+        """
+        Get the underlying FieldDesc object list that the EmbeddingColumn
+        object holds.
+
+        Returns:
+            A list of the FieldDesc objects.
+        """
+        if self.category_column is None:
+            return [FieldDesc()]
+
+        return self.category_column.get_field_desc()
+
+    def apply_to(self, field_desc):
+        """
+        Create a new EmbeddingColumn object that holds the given
+        FieldDesc object.
+
+        Args:
+            field_desc (FieldDesc): the given FieldDesc object.
+
+        Returns:
+            A new EmbeddingColumn object which holds the given
+            FieldDesc object.
+        """
+        if self.category_column is not None:
+            category_column = self.category_column.apply_to(field_desc)
+        else:
+            category_column = None
+
+        return EmbeddingColumn(category_column=category_column,
+                               dimension=self.dimension,
+                               combiner=self.combiner,
+                               initializer=self.initializer,
+                               name=self.name)
+
+
+class IndicatorColumn(FeatureColumn):
+    """
+    IndicatorColumn represents the one-hot feature column.
+
+    Args:
+        category_column (CategoryColumn): the underlying CategoryColumn object.
+        name (str): only used when category_column=None. In this case, the
+            category_column would be filled automaticaly in the feature derivation
+            stage.
+    """
+    def __init__(self, category_column=None, name=""):
+        if category_column is not None:
+            assert isinstance(category_column, CategoryColumn)
+
+        self.category_column = category_column
+        self.name = name
+
+    def get_field_desc(self):
+        """
+        Get the underlying FieldDesc object list that the IndicatorColumn
+        object holds.
+
+        Returns:
+            A list of the FieldDesc objects.
+        """
+        if self.category_column is None:
+            return [FieldDesc()]
+
+        return self.category_column.get_field_desc()
+
+    def apply_to(self, field_desc):
```

Comment:
It may be confusing that `apply_to` returns a FeatureColumn object.

Reply:
Great suggestions! Change it to be `create_feature_column_from`.